[
    {
        "user": "U053ETSL666",
        "type": "message",
        "ts": "1681638139.747009",
        "client_msg_id": "262a67ec-7210-425b-a2d1-b34d50c282dd",
        "text": "Hi there. I'm a newbie trying to use Betfairlightweight python examples for the first time. I've  succeeded  in getting passed the trading.login which generates a session token but then anything I then try with the Trading object results in this error\n\nError: {'code': -32099, 'message': 'ANGX-0007', 'data': {'APINGException': {'requestUUID': 'ie2-ang08a-prd-03211101-003751a790', 'errorCode': 'INVALID_APP_KEY', 'errorDetails': ''}, 'exceptionname': 'APINGException'}}\nFull Response: {'jsonrpc': '2.0', 'error': {'code': -32099, 'message': 'ANGX-0007', 'data': {'APINGException': {'requestUUID': 'ie2-ang08a-prd-03211101-003751a790', 'errorCode': 'INVALID_APP_KEY', 'errorDetails': ''}, 'exceptionname': 'APINGException'}}, 'id': 1}\n\nCan anyone give me any pointers?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "3c454572d52e",
            "image_72": "https:\/\/avatars.slack-edge.com\/2023-04-16\/5113998876981_3c454572d52eaeddbd41_72.png",
            "first_name": "alan",
            "real_name": "alan fisher",
            "display_name": "alan fisher",
            "team": "T4G9NBD2M",
            "name": "fishea000",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1681638139.747009",
        "reply_count": 2,
        "reply_users_count": 2,
        "latest_reply": "1681730894.345109",
        "reply_users": [
            "U02RN7YDRQ9",
            "U053ETSL666"
        ],
        "replies": [
            {
                "user": "U02RN7YDRQ9",
                "ts": "1681661127.292719"
            },
            {
                "user": "U053ETSL666",
                "ts": "1681730894.345109"
            }
        ],
        "is_locked": false,
        "subscribed": false,
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "88F",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Hi there. I'm a newbie trying to use Betfairlightweight python examples for the first time. I've  succeeded  in getting passed the trading.login which generates a session token but then anything I then try with the Trading object results in this error\n\nError: {'code': -32099, 'message': 'ANGX-0007', 'data': {'APINGException': {'requestUUID': 'ie2-ang08a-prd-03211101-003751a790', 'errorCode': 'INVALID_APP_KEY', 'errorDetails': ''}, 'exceptionname': 'APINGException'}}\nFull Response: {'jsonrpc': '2.0', 'error': {'code': -32099, 'message': 'ANGX-0007', 'data': {'APINGException': {'requestUUID': 'ie2-ang08a-prd-03211101-003751a790', 'errorCode': 'INVALID_APP_KEY', 'errorDetails': ''}, 'exceptionname': 'APINGException'}}, 'id': 1}\n\nCan anyone give me any pointers?"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "U02RN7YDRQ9",
        "type": "message",
        "ts": "1681661127.292719",
        "client_msg_id": "a2f82dce-e93d-4031-9eee-8b9f1fbd9349",
        "text": "App key error as it says on the box. <https:\/\/docs.developer.betfair.com\/display\/1smk3cen4v3lu3yomq5qye0ni\/Betting+Exceptions>\n\nDo you have a valid App key ? <https:\/\/docs.developer.betfair.com\/display\/1smk3cen4v3lu3yomq5qye0ni\/Application+Keys>",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "ge46f1f8b708",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/e46f1f8b708a630e3191de7b2c42b1d1.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0023-72.png",
            "first_name": "edblock",
            "real_name": "edblock",
            "display_name": "foxwood",
            "team": "T4G9NBD2M",
            "name": "eanb",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1681638139.747009",
        "parent_user_id": "U053ETSL666",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "NE+P",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "App key error as it says on the box. "
                            },
                            {
                                "type": "link",
                                "url": "https:\/\/docs.developer.betfair.com\/display\/1smk3cen4v3lu3yomq5qye0ni\/Betting+Exceptions"
                            },
                            {
                                "type": "text",
                                "text": "\n\nDo you have a valid App key ? "
                            },
                            {
                                "type": "link",
                                "url": "https:\/\/docs.developer.betfair.com\/display\/1smk3cen4v3lu3yomq5qye0ni\/Application+Keys"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "U053U2PUK2M",
        "type": "message",
        "ts": "1681665134.137979",
        "edited": {
            "user": "U053U2PUK2M",
            "ts": "1681665160.000000"
        },
        "client_msg_id": "badb16c3-a709-47a9-b30e-fa08740cea59",
        "text": "There is a question on the developer forum here <https:\/\/forum.developer.betfair.com\/forum\/sports-exchange-api\/exchange-api\/35113-connection-error-for-betfairlightweight> about an issue that I am also experiencing and the author was directed here.\n\nWhen logging in I am getting this error. Could anyone help resolve this?\n\n```requests.exceptions.SSLError: HTTPSConnectionPool(host='<http:\/\/identitysso-cert.betfair.com|identitysso-cert.betfair.com>', port=443): Max retries exceeded with url: \/api\/certlogin (Caused by SSLError(SSLError(9, '[SSL] PEM lib (_ssl.c:4044)')))```",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "cd650a2c5d10",
            "image_72": "https:\/\/avatars.slack-edge.com\/2023-04-16\/5114523896437_cd650a2c5d1024b02e08_72.png",
            "first_name": "eugene",
            "real_name": "eugene brown",
            "display_name": "eugene brown",
            "team": "T4G9NBD2M",
            "name": "efbbrown",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1681665134.137979",
        "reply_count": 6,
        "reply_users_count": 3,
        "latest_reply": "1685571723.660569",
        "reply_users": [
            "UBS7QANF3",
            "U053U2PUK2M",
            "U04980ZT6UX"
        ],
        "replies": [
            {
                "user": "UBS7QANF3",
                "ts": "1681717465.151159"
            },
            {
                "user": "U053U2PUK2M",
                "ts": "1681753786.704769"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1681970221.099299"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1681970445.366169"
            },
            {
                "user": "U053U2PUK2M",
                "ts": "1682108614.266449"
            },
            {
                "user": "U04980ZT6UX",
                "ts": "1685571723.660569"
            }
        ],
        "is_locked": false,
        "subscribed": false,
        "attachments": [
            {
                "from_url": "https:\/\/forum.developer.betfair.com\/forum\/sports-exchange-api\/exchange-api\/35113-connection-error-for-betfairlightweight",
                "service_icon": "https:\/\/forum.developer.betfair.com\/favicon.ico",
                "id": 1,
                "original_url": "https:\/\/forum.developer.betfair.com\/forum\/sports-exchange-api\/exchange-api\/35113-connection-error-for-betfairlightweight",
                "fallback": "Developer Program Forum: Connection error for betfairlightweight -              Developer Program Forum",
                "text": "I have been trying to use the betfairAPI (in python specifcailly using the betfairlightweight) but I haven't been successful in setting up communication yet.\n\nThe code I am trying to run is:\n&amp;quot;trading = betfairlightweight.APIClient(username=my_username,\npassword=my_password,\napp_key=my_app_key,\ncerts=certs_path)",
                "title": "Connection error for betfairlightweight -              Developer Program Forum",
                "title_link": "https:\/\/forum.developer.betfair.com\/forum\/sports-exchange-api\/exchange-api\/35113-connection-error-for-betfairlightweight",
                "service_name": "Developer Program Forum"
            }
        ],
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "4=H",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "There is a question on the developer forum here "
                            },
                            {
                                "type": "link",
                                "url": "https:\/\/forum.developer.betfair.com\/forum\/sports-exchange-api\/exchange-api\/35113-connection-error-for-betfairlightweight"
                            },
                            {
                                "type": "text",
                                "text": " about an issue that I am also experiencing and the author was directed here.\n\nWhen logging in I am getting this error. Could anyone help resolve this?\n\n"
                            }
                        ]
                    },
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "requests.exceptions.SSLError: HTTPSConnectionPool(host='identitysso-cert.betfair.com', port=443): Max retries exceeded with url: \/api\/certlogin (Caused by SSLError(SSLError(9, '[SSL] PEM lib (_ssl.c:4044)')))"
                            }
                        ],
                        "border": 0
                    }
                ]
            }
        ]
    }
]